### PR TITLE
Perf base v2

### PR DESCRIPTION
**Pull request by andyzhang2023**
### Description
1. downgrade log level from Info to Debug in eth_sendRawTransaction API.
2. optimization of truncatePending and truncateQueue
3. optimization of pendingCache disable when the node does not enable --mine.
4. split metrics of txpool by cases of 'reset' and 'reorg'.
5. add logs to trace the reset events of txpool.

### CODE REVIEW

File: core/txpool/legacypool/legacypool.go [hunk 1]
```diff
@@ -1491,20 +1514,35 @@ func (pool *LegacyPool) scheduleReorgLoop() {
...
 			reorgCount.Inc(1)
 			reorgDurationTimer.Update(reorgCost)
 			reorgNoBlockingDurationTimer.Update(reorgLockedCost)
 			promoteTimer.Update(promoteCost)
+			promoteTxCount.Inc(int64(len(promoted)))
+			reorgWaitLockTimer.Update(waittime)
+			truncatePendingTimer.Update(truncatePending)
+			truncateQueueTimer.Update(truncateQueue)
+			sendFeedTxCount.Inc(int64(sendFeedTxs))
+			feedResetTimer.Update(sendFeed)
```
Comment (ghost) [inline core/txpool/legacypool/legacypool.go]: should be feedTimer

File: core/txpool/legacypool/nonecache_for_miner.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func (nc *noneCacheForMiner) dump() map[common.Address]types.Transactions {
+	// dump all pending transactions from the pool
+	nc.pool.mu.RLock()
+	defer nc.pool.mu.RUnlock()
+	pending := make(map[common.Address]types.Transactions)
+	for addr, txlist := range nc.pool.pending {
+		pending[addr] = txlist.Flatten()
+	}
+	return pending
```
Comment (ghost) [inline core/txpool/legacypool/nonecache_for_miner.go:36]: why not do nothing

File: core/blockchain.go [hunk 3]
```diff
@@ -2547,7 +2548,7 @@ func (bc *BlockChain) SetCanonical(head *types.Block) (common.Hash, error) {
 	if timestamp := time.Unix(int64(head.Time()), 0); time.Since(timestamp) > time.Minute {
 		context = append(context, []interface{}{"age", common.PrettyAge(timestamp)}...)
 	}
-	log.Info("perf-trace Chain head was updated", context...)
```
Comment (ghost) [inline core/blockchain.go:2550]: log feed txpool here

File: core/txpool/legacypool/legacypool.go [hunk 4]
```diff
@@ -1516,10 +1554,20 @@ func (pool *LegacyPool) runReorg(done chan struct{}, reset *txpoolResetRequest,
...
+	var oldBlock, newBlock uint64 = 0, 0
 	tw := time.Now()
+	if reset != nil {
+		if reset.oldHead != nil {
+			oldBlock = reset.oldHead.Number.Uint64()
+		}
+		if reset.newHead != nil {
+			newBlock = reset.newHead.Number.Uint64()
+		}
+		log.Info("txpool-trace try to reset txpool", "oldHead", oldBlock, "newHead", newBlock)
```
Comment (ghost) [inline core/txpool/legacypool/legacypool.go:1567]: also log when resetreorg is done

File: core/txpool/legacypool/nonecache_for_miner.go (hunk 2 above)
Comment (andyzhang2023) [reply core/txpool/legacypool/nonecache_for_miner.go:36]: We need it to provide original pending transactions for other callers like RPC services.

**Pull request merged**